### PR TITLE
New CoreValue Changes

### PR DESCRIPTION
**Pull request by RighteousRaichu**
I'm waiting until Friday for some changes from Sheyne, so I'm thinking I'll merge this into master soon. Needs some cleanup work first.

### CODE REVIEW

**Comment by ilargierdi** [comment]
UI improvements are looking great! I dig the new look of the properties panel. The text on the left side of the blue bar needs a little bit of padding, since it was showing up against the left edge.

I'm not sure how much you're aware of being broken, so I'll just list what I noticed:
* position doesn't update in the graph editor when changed in the properties panel
* adder file doesn't load
* input panel wasn't working; I suspect because you're waiting for the changes from Sheyne
  - I tried testing it on the NFA and DFA, and I imagine the unit test panel would have similar results
* there's a fair bit of lag with live updating the graph editor when rapidly changing the label of a node
  - this may be an issue with the graph editor and unrelated to your changes

Unrelated to your changes, I noticed that the boundary anchor point is laggy for large nodes and boundary point calculations are incorrect for ellipses. This is more of a mental note for me to investigate when I have time. I also created #246 to reference later when I get around to it.

**Comment by Sheyne** [comment]
I'd say that the fact that `node package.js` produces executables that don't run is the only thing blocking merging this. example

**Comment by Sheyne** [comment]
So the first part of the above issue is that the build dir isn't getting copied to the application bundle anymore. This is we're because package.js hasn't really changed. If I manually copy the build dir, it also fails, but it at least shows the main GUI (console complains about missing "./plugins" dir).

**Comment by RighteousRaichu** [comment]
I'm looking into it now.

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "type": "git",
     "url": "https://github.com/2graphic/sinap-ide"
   },
-  "main": "index.js",
+  "main": "build/index.js",
```
Comment (RighteousRaichu) [inline package.json]: @Sheyne is this change necessary?

**Comment by slaymaker1907** [comment]
This is like -3000 LoC. I approve.
